processing/src/main/java/org/apache/druid/query/scan/ScanQuery.java

```diff
@@ -443,6 +443,12 @@ public List<ColumnType> getColumnTypes()
     return columnTypes;
   }
 
+  @JsonProperty("legacy")
```

Comment:
Could use a javadoc and a `@Deprecated` annotation. Javadoc could be something like:

```
/**
 * Prior to PR https://github.com/apache/druid/pull/16659 (Druid 31) data servers require
 * the "legacy" parameter to be set to a non-null value. For compatibility with older data
 * servers during rolling updates, we need to write out "false".
 */

Reply:
added, thanks :+1: